firefox-ios/Client/FeatureFlags/NimbusFlaggableFeature.swift

```diff
@@ -166,7 +161,15 @@ struct NimbusFlaggableFeature: HasNimbusSearchBar {
     public func isUserEnabled(using nimbusLayer: NimbusFeatureFlagLayer) -> Bool {
         guard let optionsKey = featureKey,
               let option = profile.prefs.boolForKey(optionsKey)
-        else { return isNimbusEnabled(using: nimbusLayer) }
+        else {
+            // In unit tests only, we provide a way to return an override value to simulate a user's preference for a feature
+            if AppConstants.isRunningUnitTest,
+               UserDefaults.standard.valueExists(forKey: PrefsKeys.NimbusUserEnabledFeatureTestsOverride) {
+                return UserDefaults.standard.bool(forKey: PrefsKeys.NimbusUserEnabledFeatureTestsOverride)
+            }
```

Comment:
I basically just hijacked some existing code and moved it down into the `isUserEnabled()` method instead of `isNimbusEnabled()`, so it only overrides the user's preference.

I had to move this code because `isNimbusEnabled()` also gets called in `getNimbusOrDebugSetting()`, which was overriding both the nimbus flag _and_ the user preference flag in our unit tests... We don't need to override the nimbus flag though because we have better ways of doing that (e.g. https://github.com/mozilla-mobile/firefox-ios/pull/23725#discussion_r1882718322).